functions/visualizations.py

```diff
@@ -79,9 +79,88 @@ def _get_transforms() -> dict:
     return {
         "clustergrammer": _ClustergrammerTransform(),
         "ihc_combined_plot": _ihc_combined_transform,
+        "additional_metadata": _add_antibody_metadata,
     }
 
 
+def _add_antibody_metadata(
+    file_record: DownloadableFiles, metadata_df: pd.DataFrame
+) -> Optional[dict]:
+    """
+    Pseudo transformation to add antibody data to the DownloadableFiles.additional_metadata JSON
+    Only for upload_type in [cytof, elisa, ihc, micsss, and mif]
+    """
+    upload_type = file_record.upload_type.lower()
+    if upload_type not in ["cytof", "elisa", "ihc", "micsss", "mif"]:
+        return None
+
+    file_md = file_record.additional_metadata
+
+    ct_md = TrialMetadata.find_by_trial_id(file_record.trial_id).metadata_json
+    assay_md = ct_md.get("assays", {}).get(file_record.upload_type.lower(), {})
+
+    # differences in data model
+    if upload_type == "cytof":
+        key = "cytof_antibodies"
+    elif upload_type in ["elisa", "mif"]:
+        key = "antibodies"
+    elif upload_type in ["ihc", "micsss"]:
+        key = "antibody"
```

Comment:
nit: having separate helper functions for each of these upload types might make the code a bit easier to follow, even though that'd be slightly more verbose.

Reply:
in next commit